examples/model_bertabs.py

```diff
@@ -0,0 +1,719 @@
+from collections import namedtuple
+import copy
+import math
+import numpy as np
+import torch
+from torch import nn
+from torch.nn.init import xavier_uniform_
+import pdb
+from transformers import BertModel, BertConfig, PreTrainedEncoderDecoder
+
+
+MAX_SIZE = 5000
+
+
+BertAbsConfig = namedtuple(
+    "BertAbsConfig",
+    ["temp_dir", "large", "finetune_bert", "encoder", "share_emb", "max_pos", "enc_layers", "enc_hidden_size", "enc_heads", "enc_ff_size", "enc_dropout", "dec_layers", "dec_hidden_size", "dec_heads", "dec_ff_size", "dec_dropout"],
+)
+
+
+def get_bertabs(path_to_checkpoints):
+    config = BertAbsConfig(
+        temp_dir=".",
+        finetune_bert=False,
+        large=False,
+        share_emb=True,
+        encoder="bert",
+        max_pos=512,
+        enc_layers=6,
+        enc_hidden_size=512,
+        enc_heads=8,
+        enc_ff_size=512,
+        enc_dropout=0.2,
+        dec_layers=6,
+        dec_hidden_size=768,
+        dec_heads=8,
+        dec_ff_size=2048,
+        dec_dropout=0.2,
+    )
+    checkpoints = torch.load(path_to_checkpoints, lambda storage, loc: storage)
+    bertabs = BertAbsSummarizer.from_pretrained(checkpoints, config, torch.device("cpu"))
+    bertabs.eval()
+    return bertabs
+
+
+class BertAbsSummarizer(nn.Module):
+    def __init__(self, args, device, checkpoint=None, bert_extractive_checkpoint=None):
+        super(BertAbsSummarizer, self).__init__()
+        self.args = args
+        self.device = device
+        self.encoder = Bert(args.large, args.temp_dir, args.finetune_bert)
+
+        # If pre-trained weights are passed for Bert, load these.
+        load_bert_pretrained_extractive = True if bert_extractive_checkpoint else False
+        if load_bert_pretrained_extractive:
+            self.encoder.model.load_state_dict(
+                dict([(n[11:], p) for n, p in bert_extractive_checkpoint.items() if n.startswith('bert.model')]), strict=True)
+
+        if (args.encoder == 'baseline'):
+            bert_config = BertConfig(self.encoder.model.config.vocab_size, hidden_size=args.enc_hidden_size,
+                                     num_hidden_layers=args.enc_layers, num_attention_heads=8,
+                                     intermediate_size=args.enc_ff_size,
+                                     hidden_dropout_prob=args.enc_dropout,
+                                     attention_probs_dropout_prob=args.enc_dropout)
+            self.encoder.model = BertModel(bert_config)
+
+        if(args.max_pos > 512):
```

Comment:
whitespace

Reply:
Probably. I’m not changing the authors my code formatting for now, as it makes the comparison easier. Will run everything through `black` before merging though.